nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi.sh

```diff
@@ -265,20 +265,54 @@ run() {
     BOOTSTRAP_DIR_PARAMS="${BOOTSTRAP_LOG_PARAMS} ${BOOTSTRAP_PID_PARAMS} ${BOOTSTRAP_CONF_PARAMS}"
 
     RUN_NIFI_CMD="cd "\""${NIFI_HOME}"\"" && ${sudo_cmd_prefix} "\""${JAVA}"\"" -cp "\""${BOOTSTRAP_CLASSPATH}"\"" -Xms12m -Xmx24m ${BOOTSTRAP_DIR_PARAMS}  org.apache.nifi.bootstrap.RunNiFi"
+    LSB_EXIT_STATUS=0
+    LSB_NIFI_STATUS_TEMPFILE=`mktemp -t LSB_NIFI_STATUS_TEMPFILE.XXXXXXXXXX`
 
     if [ "$1" = "start" ]; then
         (eval $RUN_NIFI_CMD $@ &)
```

Comment:
Hi @apiri,
I could not find a proper way to meet that part of spec due to async nature
of startup mechanism.
The best tradeoff that came in my mind is to run the status action again
after that 3 sec sleep time.
As the current problem stated by @evega is more to do with status call from
puppet etc. I had left that startup part untouched.
Any guidance will be greatly appreciated.

Thanks & Regards,
Puspendu Banerjee

On Jul 26, 2016 8:02 AM, "Aldrin Piri" <notifications@github.com> wrote:

In
nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi.sh
<https://github.com/apache/nifi/pull/675#discussion_r72243818>:

> @@ -265,20 +265,54 @@ run() {
>      BOOTSTRAP_DIR_PARAMS="${BOOTSTRAP_LOG_PARAMS} ${BOOTSTRAP_PID_PARAMS} ${BOOTSTRAP_CONF_PARAMS}"
>
>      RUN_NIFI_CMD="cd "\""${NIFI_HOME}"\"" && ${sudo_cmd_prefix} "\""${JAVA}"\"" -cp "\""${BOOTSTRAP_CLASSPATH}"\"" -Xms12m -Xmx24m ${BOOTSTRAP_DIR_PARAMS}  org.apache.nifi.bootstrap.RunNiFi"
> +    LSB_EXIT_STATUS=0
> +    LSB_NIFI_STATUS_TEMPFILE=`mktemp -t LSB_NIFI_STATUS_TEMPFILE.XXXXXXXXXX`
>
>      if [ "$1" = "start" ]; then
>          (eval $RUN_NIFI_CMD $@ &)

This seems like we would not be fulfilling the spec if nifi did not start
appropriately. I will get a 0 regardless if it starts successfully or not.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/apache/nifi/pull/675/files/5cc306c843b4056845eb483b9581a6b71c326dfc#r72243818>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AE2raCAgQXyFIiTNu8aZLoEHgwWyTF2Yks5qZgVfgaJpZM4JP4_d>
.